src/App.tsx

```diff
@@ -15,37 +17,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SetByReverse = true;
+
+enum SortedFields {
+  SetById,
+  SetByAlphabet,
+  SetByLength,
+  SetByDanger,
+}
+
+function getSortedGoods(
+  goods: string[],
+  { sortField, reversed }: { sortField: SortedFields; reversed: boolean },
+) {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (sortField) {
+      case SortedFields.SetById:
+      case SortedFields.SetByAlphabet:
+        return good1.localeCompare(good2);
+      case SortedFields.SetByLength:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setsortField] = useState(SortedFields.SetByDanger);
```

Comment:
The initial state for 'sortField' is set to 'SortedFields.SetByDanger', but there is no case in the 'getSortedGoods' function that handles this sorting field. Either implement the sorting logic for 'SetByDanger' or adjust the initial state to a valid sorting option.